src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +101,11 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        CandidateValidator candidateValidator = new CandidateValidator();
+        return candidates.stream()
+                .filter(c -> candidateValidator.test(c))
```

Comment:
```suggestion
        return candidates.stream()
                .filter(new CandidateValidator())
```
I'd recommend something like this, as this is more concise